app/src/androidTest/java/com/github/geohunt/app/maps/GoogleMapDisplayTest.kt

```diff
@@ -24,38 +33,109 @@ class GoogleMapDisplayTest {
     @get:Rule
     val composeTestRule = createComposeRule()
 
+    @get:Rule
+    val fineLocationPermissionRule: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+    @get:Rule
+    val coarseLocationPermissionRule: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_COARSE_LOCATION)
+
     private val epflCoordinates = LatLng(46.51958, 6.56398)
     private var mockTestChallengeDatabase = mutableListOf<Marker>()
 
+    private val mockLocation = Location(46.51958, 6.56398)
+    private val mockLocationRepo = object : LocationRepositoryInterface {
+        override fun getLocations(coroutineScope: CoroutineScope): Flow<Location> {
+            return flowOf(mockLocation)
+        }
+    }
+
+    val mockRepositoryChallenge = Challenge(
+        id = "Event 3",
+        authorId = "test",
+        photoUrl = "https://picsum.photos/300/300",
```

Comment:
I see your concern. It is bad to have implicit dependencies. Nevertheless, these dependencies did not have impact on the functioning of our application.

I have removed the dependencies.

Reply:
These are tests so of course there is no impact on our application. Now imagine 2 years in the future picsum stop working and suddenly all of our tests crashes. It would be a hell of a day to figure out the fuck is happening. 

Thanks anyway

Reply:
What I meant is that the tests were designed in such a way that even if the picsum pictures would stop working, the tests would still pass.